docs/src/main/asciidoc/writing-extensions.adoc

```diff
@@ -1238,6 +1238,130 @@ HealthBuildItem addHealthCheck(AgroalBuildTimeConfig agroalBuildTimeConfig) {
+
+==== Case 1: The library uses MP Metrics
+If the library exposes metrics by itself, we don't have to do much. However, there are a few points to consider:
+
+- It should be possible to disable all metrics for an extension, which can be a bit problematic if the library registers them directly
+rather than through the unified registration mechanism in the `quarkus-smallrye-metrics-spi` module. Therefore the library should contain
+a way to turn all metrics off, for example using a library-specific system property that can be set during build time (by emitting a `SystemPropertyBuildItem`),
+and will disable metrics if at least one of the properties `quarkus.<extensionName>.metrics.enabled` and `quarkus.smallrye-metrics.extensions.enabled` is `false`.
+- It is desirable to be able to omit the MP Metrics dependency at runtime, so if possible, the library should be written in a way that
+it will still work when the MP Metrics dependencies (or at least the implementation, `io.smallrye:smallrye-metrics`) are unavailable. This can be achieved by
+wrapping all code that does something metric-related into an `if` condition that checks whether metrics integration is enabled. If the library performs injections
+of the `MetricRegistry`, which is not avoidable by introducing an `if` condition, it will unfortunately not be possible to remove the `microprofile-metrics` dependency,
+but it is still possible to avoid the need for `smallrye-metrics` (which contains the implementation class of the metric registry) by introducing a custom "no-op"
+implementation of the `MetricRegistry` which can, for example, return `null` from all its methods. This no-op implementation should be added to the application's classes
+so that it can be injected instead of the regular `io.smallrye.metrics.MetricRegistryImpl`.
+
+==== Case 2: The library provides its own metric API
+In this case, the extension can make use of the `quarkus-smallrye-metrics-spi` module to expose the metrics from the library using MP Metrics.
+
+The extension author should to the following:
+
+- Import the `quarkus-smallrye-metrics-spi` library in your deployment module.
+- Import the `quarkus-smallrye-metrics` extension as an **optional** dependency in your runtime module so it will not impact the size of the application if
+metrics are not included.
+- In your processor, produce any number of `MetricBuildItem` items as shown in the example below.
+The metric name of all metrics should start with a short name of the extension that owns them.
+- Add a way to disable the extension's metrics via a config item `quarkus.<extension>.metrics.enabled` that should be enabled by default. The value
+of this property will be passed to all `MetricBuildItem` objects that the extension produces and if it is false, the metric will be ignored.
+
+An example build step for build-time registration of metrics:
+
+[source%nowrap.java]
+----
+@BuildStep
```

Comment:
@lhauspie, what's your take on this? IIUC, it'd require the KafkaStreams extension to list all the metrics explicitly, instead of the generic implementation we currently have (in the blog post). It seems that'd be quite a bit more work. WDYT?